### PR TITLE
Add menu bar to CCP dashboard

### PR DESCRIPTION
**Pull request by JasonYG**
[Notion ticket](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3751aa62a2594c74b7e0ecb363304c24&p=ff794716beac4d77a0274c00436e9017)

## Changes
- Added `MenuAppBar` to `CCPDashboardPage` component
- Added new optional prop, `selectedCcp`, to `MenuAppBar` to highlight the current CCP
- Added new class to highlight current CCP in `MenuAppBar`
- Changed padding of "Hospital Overview" tab bar to be aligned with the hospitals box underneath

## Screenshots
New CCP dashboard screen with menu bar and aligned bar underneath
image

Highlighted current CCP in MenuAppBar
image

## Testing
- Check that the MenuAppBar opens properly on CCP dashboard pages
- Check that the current CCP is highlighted correctly
- Verify that the MenuAppBar isn't broken for the EventDashboardPage, when a CCP isn't selected

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] fill out `Changes`
  - [x] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/components/CCPDashboard/CCPDashboardPage.tsx
Comment (JasonYG) [inline src/components/CCPDashboard/CCPDashboardPage.tsx]: this aligns the "Hospital Overview" tab with the hospitals box underneath

File: src/components/CCPDashboard/CCPDashboardPage.tsx
Comment (JasonYG) [inline src/components/CCPDashboard/CCPDashboardPage.tsx]: notice that the new prop is being passed in

File: src/components/common/MenuAppBar.tsx
Comment (JasonYG) [inline src/components/common/MenuAppBar.tsx:28]: the new prop is optional, so it shouldn't affect current instances of the component

File: src/components/common/MenuAppBar.tsx
Comment (JasonYG) [inline src/components/common/MenuAppBar.tsx:146]: adds highlighting for the currently selected CCP

**Comment by petorsong** [comment]
works as expected but the styling is super wonky on ipad size - i'll follow up with rishabh about how we want to test screen sizes and get back to you for tomorrow; i may include additional styling changes as part of this PR
image

**Review by aaronabraham311 — CHANGES_REQUESTED** [review]
Looks great but this does not conform to our Figma designs as attached below. Specifically, Figma specifies that the menu bar should be titled with the CCP's name as well as the location. I think this should be a relatively easy change though. 
Screen Shot 2020-10-07 at 7 12 12 PM

**Comment by JasonYG** [comment]
> works as expected but the styling is super wonky on ipad size - i'll follow up with rishabh about how we want to test screen sizes and get back to you for tomorrow; i may include additional styling changes as part of this PR
> image

Yeah for sure, let me know if there's any styling changes you'd like me to implement

**Comment by JasonYG** [comment]
> Looks great but this does not conform to our Figma designs as attached below. Specifically, Figma specifies that the menu bar should be titled with the CCP's name as well as the location. I think this should be a relatively easy change though.
> Screen Shot 2020-10-07 at 7 12 12 PM

Good catch, totally missed the additional text in the menu bar.

**Review by carelynntsai — COMMENTED** [review]
Works for me! I have nothing to add on to Peter and Aaron's comments

**Comment by carelynntsai** [comment]
> > Looks great but this does not conform to our Figma designs as attached below. Specifically, Figma specifies that the menu bar should be titled with the CCP's name as well as the location. I think this should be a relatively easy change though.
> > Screen Shot 2020-10-07 at 7 12 12 PM
> 
> Good catch, totally missed the additional text in the menu bar.

Also just wondering if the refresh button on the right side of the nav bar is part of this ticket, this was already implemented for  the resource management dashboard so it should be easy to copy over

**Comment by JasonYG** [comment]
> > > Looks great but this does not conform to our Figma designs as attached below. Specifically, Figma specifies that the menu bar should be titled with the CCP's name as well as the location. I think this should be a relatively easy change though.
> > > Screen Shot 2020-10-07 at 7 12 12 PM
> > 
> > 
> > Good catch, totally missed the additional text in the menu bar.
> 
> Also just wondering if the refresh button on the right side of the nav bar is part of this ticket, this was already implemented for the resource management dashboard so it should be easy to copy over

oh yeah good catch, @petorsong thoughts here?

File: src/components/CCPDashboard/CCPDashboardPage.tsx
Comment (JasonYG) [inline src/components/CCPDashboard/CCPDashboardPage.tsx:149]: hard-coded for now; @petorsong hope this is ok

File: src/components/CCPDashboard/CCPDashboardPage.tsx
Comment (JasonYG) [inline src/components/CCPDashboard/CCPDashboardPage.tsx:143]: this jsx element is passed to MenuBar, which renders it as the title

File: src/components/common/MenuAppBar.tsx
Comment (JasonYG) [inline src/components/common/MenuAppBar.tsx:27]: added React.ReactNode so I can pass in a React element directly. This just made it easier to add the location icon, without screwing up current uses of the component

File: src/components/CCPDashboard/CCPDashboardPage.tsx
Comment (petorsong) [reply src/components/CCPDashboard/CCPDashboardPage.tsx:149]: yep - any location/address related item for CCPs will be covered under @aaronabraham311 's maps/location work

**Review by aaronabraham311 — APPROVED** [review]
Looks good!

**Review by aaronabraham311 — DISMISSED** [review]
Looks good but there is a minor issue that I saw: the location pin doesn't seem aligned with the text. 

Screen Shot 2020-10-16 at 8 51 12 PM

**Review by carelynntsai — DISMISSED** [review]
image
Looks good! Still doesn't include the refresh button but I'm good to approve it if it's not part of this ticket @petorsong

**Comment by petorsong** [comment]
let's not add the refresh button in with this PR - it can be bundled together with @carelynntsai 's RTU stuff since the behaviour will be better defined there

**Comment by JasonYG** [comment]
> Looks good but there is a minor issue that I saw: the location pin doesn't seem aligned with the text.
> 
> Screen Shot 2020-10-16 at 8 51 12 PM

Good catch, CSS is a nightmare man I should just switch to data science. Fixed in latest push

**Review by petorsong — DISMISSED** [review]
good to go - if there are any outstanding styling issues let's hold off for now and tackle that after product has taken a look, i'd like to prioritize just getting this menu bar merged in

**Review by aaronabraham311 — APPROVED** [review]
Looks good!

**Review by carelynntsai — APPROVED** [review]
🍕🍕🍕🍕🍕

**Pull request merged**
